apps/docs/pages/guides/getting-started/tutorials/with-kotlin.mdx

```diff
@@ -0,0 +1,816 @@
+                postgrest["products"].insert(productDto)
+                true
+            }
+            true
+        } catch (e: java.lang.Exception) {
+            throw e
+        }
+    }
+
+    override suspend fun getProducts(): List<ProductDto>? {
+        return withContext(Dispatchers.IO) {
+            val result = postgrest["products"]
+                .select().decodeList<ProductDto>()
+            result
+        }
+    }
+
+
+    override suspend fun getProduct(id: String): ProductDto {
+        return withContext(Dispatchers.IO) {
+            postgrest["products"].select {
+                eq("id", id)
+            }.decodeSingle<ProductDto>()
+        }
+    }
+
+    override suspend fun deleteProduct(id: String) {
+        return withContext(Dispatchers.IO) {
+            postgrest["products"].delete {
+                eq("id", id)
+            }
+        }
+    }
+
+    override suspend fun updateProduct(
+        id: String,
+        name: String,
+        price: Double,
+        imageName: String,
+        imageFile: ByteArray
+    ) {
+        withContext(Dispatchers.IO) {
+            if (imageFile.isNotEmpty()) {
+                val imageUrl =
+                    storage["Product%20Image"].upload(
+                        path = "$imageName.png",
+                        data = imageFile,
+                        upsert = true
+                    )
+                postgrest["products"].update({
+                    set("name", name)
+                    set("price", price)
+                    set("image", buildImageUrl(imageFileName = imageUrl))
+                }) {
+                    eq("id", id)
+                }
+            } else {
+                postgrest["products"].update({
+                    set("name", name)
+                    set("price", price)
+                }) {
+                    eq("id", id)
+                }
+            }
+        }
+    }
+
+    // Because I named the bucket as "Product Image" so when it turns to an url, it is "%20"
+    // For better approach, you should create your bucket name without space symbol
+    private fun buildImageUrl(imageFileName: String) =
+        "${BuildConfig.SUPABASE_URL}/storage/v1/object/public/${imageFileName}".replace(" ", "%20")
+}
+```
+
+### 8. Implement screens
+<Admonition type="note">
+  By the time this page is written, there might be some changes in the code to make it more simple. If you face any problem when implement, feel free to refer the example repository above
+</Admonition>
+
+#### 8.1 Product list
+Create ProductListViewModel
+```kotlin
+@HiltViewModel
+class ProductListViewModel @Inject constructor(
+    private val productRepository: ProductRepository,
+) : ViewModel() {
+
+    private val _productList = MutableStateFlow<List<Product>?>(listOf())
+    val productList: Flow<List<Product>?> = _productList
+
+
+    private val _isLoading = MutableStateFlow(false)
+    val isLoading: Flow<Boolean> = _isLoading
```

Comment:
Can we only show part of the code block with collapse and expand button? 

cc @gregnr

Reply:
@phamhieu Unfortunately no built-in expand functionality atm (but good idea!). Might be able to achieve something similar with one of Code Hike's components like [Spotlight](https://codehike.org/demo/spotlight). I would probably opt to leave this as is for now until we have a good pattern for this.